packages/http/src/__tests__/client.spec.ts

```diff
@@ -0,0 +1,55 @@
+import * as nock from 'nock';
+import createNewClientInstance, { PrismHttp } from '../client';
+import { forwarder } from '../forwarder';
+import { mocker } from '../mocker';
+
+type PromiseType<T> = T extends Promise<infer U> ? U : T;
+
+describe('User Http Client', () => {
+  afterEach(() => nock.cleanAll());
+
+  describe('with mocking set to false', () => {
+    describe('get a resource', () => {
+      let response: PromiseType<ReturnType<PrismHttp['get']>>;
+      beforeAll(async () => {
+        jest.spyOn(forwarder, 'fforward');
+        jest.spyOn(mocker, 'mock');
+
+        nock('https://petstore.swagger.io/v2')
+          .get('/pet/10')
+          .reply(200, {
+            license: {
+              key: 'mit',
+              name: 'MIT License',
+              spdx_id: 'MIT',
+              url: 'https://api.github.com/licenses/mit',
+              node_id: 'MDc6TGljZW5zZTEz',
+            },
+          });
+
+        const client = await createNewClientInstance(
+          { mock: false, validateRequest: true, validateResponse: true },
+          require.resolve('../../../../examples/petstore.oas2.yaml'),
```

Comment:
Can we avoid using examples for tests? Let’s duplicate the file and make a fixture instead.

Reply:
Definitely — this is just to show how the Client/API will look like. I'll finalise this when we're good with it.